website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,281 @@
+
+```shell-session
+$ kubectl get HTTPRoute,TCPRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )} '
+```
+
+Note, the above command will retrieve all HTTPRoutes and TCPRoutes. TLSRoutes and UDPRoutes are not supported in v0.1.0.
+
+If your output is empty, you can skip to the last step; otherwise, you have existing routes that require a new ReferencePolicy. In this case, your output will appear similar to the following:
+
+```log
+{
+  "name": "example-http-route",
+  "namespace": "example-namespace",
+  "kind": "HTTPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+{
+  "name": "example-tcp-route",
+  "namespace": "a-different-namespace",
+  "kind": "TCPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+```
+
+**3.** Create a ReferencePolicy to allow cross namespace traffic for each route service pair.
+
+Using the list of routes you created earlier as a guide, create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy) to allow cross namespace traffic.
+You will need to create a reference policy that explicitly allows each cross namespace route to service pair to prevent the route from breaking. If you've already created a ReferencePolicy you can skip this step.
+<!---
+TODO: add link to our docs on Cross Namespace Reference Policies, once we have written then, and tell the user to see them for more details on how to create these policies.
+--->
+For example, the above output would require a reference policy that looks as follows.
+
+Note, the ReferencePolicy should be created in the same namespace that it is allowing traffic to
+
+<CodeBlockConfig filename="referencepolicy.yaml">
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferencePolicy
+metadata:
+  name: reference-policy
+  namespace: gateway-namespace
+spec:
+  from:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      namespace: example-namespace
+    - group: gateway.networking.k8s.io
+      kind: TCPRoute
+      namespace: a-different-namespace
+  to:
+    - group: ""
+      kind: Service
+      name: web-backend
+</div>
+```
+</CodeBlockConfig>
+
+Edit your ReferencePolicy so all routes are allowed and then save into a file called referencepolicy.yaml
+
+Run the following command to apply it to your cluster
+
+```shell-session
+$ kubectl apply -f referencepolicy.yaml
```

Comment:
We generally want to use the longer version of flag names everywhere to help out anyone reading the docs that isn't super familiar with kubectl

```suggestion
$ kubectl apply --filename referencepolicy.yaml
```